### PR TITLE
feat: enable arm64 on AWS

### PR DESCRIPTION
**Pull request by ezr-ondrej**
This should enable arm64, it's not perfect for sure, but at least there's a test! :stuck_out_tongue: 

Refs HMS-1135

### CODE REVIEW

File: internal/services/aws_reservation_service.go [hunk 1]
```diff
@@ -52,19 +53,16 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Validate architecture match (hardcoded since image builder currently only supports x86_64). This can be only done
-	// when launch template is not set.
-	if payload.LaunchTemplateID == "" {
```
Comment (ezr-ondrej) [inline internal/services/aws_reservation_service.go:57]: :warning: I did not undesrtand why, so I've removed the condition for LaunchTemplate being empty :thinking:

File: internal/clients/http/errors.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ var (
 
 // Image Builder
 var (
+	ErrImageArchInvalid     = usrerr.New(500, "the image provided has invalid architecture and can not be launch with provided instance type", "invalid image architecture")
```
Comment (lzap) [inline internal/clients/http/errors.go:17]:
```suggestion
	ErrImageArchInvalid     = usrerr.New(400, "the image provided has invalid architecture and can not be launch with provided instance type", "invalid image architecture")
```

File: internal/services/aws_reservation_service.go [hunk 3]
```diff
@@ -52,19 +53,16 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
...
+	supportedArch := "x86_64"
+	it := preload.EC2InstanceType.FindInstanceType(clients.InstanceTypeName(payload.InstanceType))
+	if it == nil {
+		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), fmt.Sprintf("unknown type: %s", payload.InstanceType), ErrUnknownInstanceTypeName))
+		return
+	}
+	if it.Architecture.String() != supportedArch && it.Architecture.String() != "arm64" {
+		renderError(w, r, payloads.NewWrongArchitectureUserError(r.Context(), ErrArchitectureMismatch))
+		return
 	}
```
Comment (lzap) [inline internal/services/aws_reservation_service.go:66]: Can you simply make `supportedArch` a slice and simplify the condition so we can easily add more later? This looks clunky now.

**Review by lzap — CHANGES_REQUESTED** [review]
Just a small nickpick, nice.

File: internal/clients/http/errors.go (hunk 2 above)
Comment (ezr-ondrej) [reply internal/clients/http/errors.go]: Eh, totally correct! Thank you! :+1:

File: internal/services/aws_reservation_service.go (hunk 3 above)
Comment (ezr-ondrej) [reply internal/services/aws_reservation_service.go:67]: Thank you, it was totally just my lazieness, thanks, fixed :+1:

**Comment by ezr-ondrej** [comment]
> Ok, tests are failing tho.

it was an regression! :facepalm: Fixed :)

**Comment by lzap** [comment]
Thanks.

**Pull request merged**